### ISSUE TITLE
Sync roles with multiple User classes

### ISSUE THREAD

**Issue by venomtm** [JosephSilber/bouncer#458]
I have multiple User-like classes that use the trait HasRolesAndAbilities.

This results in table data like:
```
+----+---------+-----------+----------------+...
| id | role_id | entity_id | entity_type    |...
+----+---------+-----------+----------------+...
|  3 |       2 |         1 | App\Member     |...
|  1 |       1 |         1 | App\User       |...
|  4 |       9 |         2 | App\Member     |...
|  2 |       1 |         2 | App\User       |...
+----+---------+-----------+----------------+...
```

When using `Bouncer::sync` to sync roles for users or members the entity_type is not respected which causes the other rows to be removed.

This does not work:
`Bouncer::sync($user)->roles(['admin']);`

This does work:
`$user->roles()->sync([1]);`

**Comment by venomtm** [comment]
This might be related to: https://github.com/JosephSilber/bouncer/issues/409

**Comment by JosephSilber** [comment]
Is this still an issue in [v1.0.0-rc.6](https://github.com/JosephSilber/bouncer/releases/tag/v1.0.0-rc.6)?

It fixes the issue #409 that you mentioned.

**Comment by venomtm** [comment]
Hi,

Sorry for the late reply!

Unfortunately this still doesn't work in v1.0.0-rc.6. I have not tested v1.0.0-rc.7 though.

Regards, Mikael
